src/BulkIssueCreatorCLI/Program.cs

```diff
@@ -0,0 +1,355 @@
+        private static readonly Settings s_settings = new Settings();
+        private static IssueManager s_issueManager;
+        private static Arguments s_arguments;
+        static void Main(string[] args)
+        {
+            using FileLogWriter fileLogger = new FileLogWriter("IssueCreator.log");
+            Colorizer.SetupWriter(fileLogger);
+            if (!Parser.TryParse<Arguments>(args, out s_arguments))
+            {
+                return;
+            }
+
+            s_logger = new FileLogger(Path.Combine(Settings.SettingsFolder, "bulkIssueCreatorCLI.log"));
+
+            if (s_arguments.Type == ActionType.createIssues)
+            {
+                CreateIssuesActionAsync().GetAwaiter().GetResult();
+            }
+        }
+
+        private static async Task CreateIssuesActionAsync()
+        {
+            using LoggingScope main = new LoggingScope("Create issues in bulk on GitHub");
+
+            using (LoggingScope li = new LoggingScope("Loading settings..."))
+            {
+                using (IDisposable scope = s_logger.CreateScope("Loading settings"))
+                {
+                    s_settings.Initialize(Settings.Deserialize(Settings.SettingsFile, s_logger));
+
+                    s_issueManager = IssueManager.Create(s_settings, ".", s_logger);
+                }
+            }
+
+            string progressFile = new FileInfo("progress.dat").FullName;
+
+            List<IssueToCreateWithEpic> issuesToCreate = null;
+            // Do we have a progress file?
+            if (File.Exists(progressFile))
+            {
+                Colorizer.WriteLine("Found progress file [Yellow!{0}]. Do you want to use it[Green!y]/[Red!n]", progressFile);
+                if (Console.ReadKey().Key == ConsoleKey.Y)
+                {
+                    // load the data from file.
+                    issuesToCreate = LoadIssuesFromProgressFile(progressFile);
+                }
+            }
+
+            if (issuesToCreate == null)
+            {
+                // if we did not load any issues from the progress file
+                using (LoggingScope li = new LoggingScope("Processing input file [Yellow!{0}]", s_arguments.InputFile))
+                {
+                    // Parse the issue data from the input file
+                    issuesToCreate = ParseInputFile(new CsvConfiguration(CultureInfo.InvariantCulture, missingFieldFound: null));
+                }
+
+                using (LoggingScope li = new LoggingScope("Identifying issues with existing parent epics."))
+                {
+                    IdentifyIssueDependencies(issuesToCreate);
+                }
+
+                using (LoggingScope li = new LoggingScope("Validate and set milestone data"))
+                {
+                    // Validate the milestone data (and create the milestone objects as needed)
+                    await ValidateAndSetMilestonesAsync(issuesToCreate).ConfigureAwait(false);
```

Comment:
Tip: don't really need to configure the awaiter in an app itself unless you know there's a good reason to.

Reply:
That was copied from the WinForms side of the app I believe. It isn't needed here.